src/workerd/util/sqlite.h

```diff
@@ -19,6 +20,20 @@ namespace workerd {
 using kj::byte;
 using kj::uint;
 
+class SqliteObserver {
+public:
+  virtual void addRowsRead(uint64_t rowsRead) {}
+  virtual void addRowsWritten(uint64_t rowsWritten) {}
+  virtual void setSrsStoredBytesCallback(kj::Function<uint64_t()> callback) {}
+
+  virtual ~SqliteObserver() noexcept(false) {}
```

Comment:
You don't need to declare this desturctor. In KJ code, `kj::Own` separately keeps track of how to dispose the object, so the destructor is never called directly on a virtual interface. Declaring the destructor virtual will actually result in a slight performance hit without any benefit for KJ code.

Note if you get a compiler warning about non-virtual destructors, the solution is to mark the derived class as `final`.